Rover/sailboat.cpp

```diff
@@ -260,6 +264,58 @@ void Sailboat::get_throttle_and_mainsail_out(float desired_speed, float &throttl
         wingsail_out *= -1.0f;
     }
 
+    //
+    // direct mast rotation control
+    //
+
+    if (!is_positive(desired_speed)) {
+        // rotating sails can be used to reverse, but not in this version
+        mast_rotation_out = 0.0f;
+    } else {
+        // + is wind over starboard side, - is wind over port side
+        float wind_dir_apparent_signed = degrees(rover.g2.windvane.get_apparent_wind_direction_rad());
+        float wind_dir_apparent_abs = fabsf(wind_dir_apparent_signed);
+        float wind_dir_apparent_sign = is_equal(wind_dir_apparent_abs, 0.0f) ? 0.0f : wind_dir_apparent_signed / wind_dir_apparent_abs;
+
+        // compute absolute sail rotation
+        float mast_rotation_angle = 0.0f;
+        bool mast_rotation_necessary = true;
+        if (wind_dir_apparent_abs < sail_angle_ideal) {
+            // in irons, center the sail.
+            mast_rotation_angle = 0.0f;
+        } else if (wind_dir_apparent_abs < (90.0f + sail_angle_ideal)) {
```

Comment:
Should this 90 be sail_angle_max?

Reply:
No. This 90 is an invariant of my sail control algorithm. `sail_angle_max` is an artifact of user setup. As my comment on `@Param: ANGLE_MAX` says, you can have a setup where the linkage is capable of rotating the mast more than +-90, but the code will limit it to 90.

Reply:
Conceivably, the use may set it up so that less than +-90 is achievable. This is suboptimal, but will still result in correct AoAs on upwind headings.